images/vmi-router/netlinkmanager/manager.go

```diff
@@ -275,31 +278,23 @@ func getCiliumInternalIPAddress(node *ciliumv2.CiliumNode) string {
 	return ""
 }
 
-func (m *Manager) DeleteRoute(vm *virtv1alpha2.VirtualMachine) {
-	// Check if IP is in cache. Do not delete routes for unknown IPs.
-	vmKey := fmt.Sprintf("%s/%s", vm.GetNamespace(), vm.GetName())
-
-	// Get IP either from Status, or from cache.
-	var vmIP string
-	if vm != nil {
-		vmIP = vm.Status.IPAddress
-	}
+func (m *Manager) DeleteRoute(vmKey types.NamespacedName, vmIP string) {
 	if vmIP == "" {
 		// Try to recover IP from the cache.
 		m.vmIPsLock.RLock()
-		vmIP = m.vmIPs[vmKey]
+		vmIP = m.vmIPs[vmKey.String()]
```

Comment:
It is safer to put elements in the map, also using the String() method from the types.NamespacedName, so as not to duplicate the logic of obtaining the key.